openedx/core/djangoapps/content/block_structure/config/__init__.py

```diff
@@ -10,12 +10,51 @@
 from .models import BlockStructureConfiguration
 
 # Switches
+# .. toggle_name: block_structure.invalidate_cache_on_publish
+# .. toggle_implementation: WaffleSwitch
+# .. toggle_default: False
+# .. toggle_description: When enabled, the block structure cache is invalidated when changes to
+#   courses are published. Enabling this toggle is useful for non-expiring block structure cache.
+#   It is also useful to enable this when `block_structure.storage_backing_for_cache` is enabled
+#   because it guarantees the storage, containing the storage-backed structures, is updated with
+#   the latest changes.
+# .. toggle_use_cases: opt_in
+# .. toggle_creation_date: 2017-02-23
+# .. toggle_tickets: https://github.com/edx/edx-platform/pull/14571
 INVALIDATE_CACHE_ON_PUBLISH = WaffleSwitch(
     "block_structure.invalidate_cache_on_publish", __name__
 )
+
+# .. toggle_name: block_structure.storage_backing_for_cache
+# .. toggle_implementation: WaffleSwitch
+# .. toggle_default: False
+# .. toggle_description: When enabled, block structures are stored in a more permanent storage,
+#   like a database, which provides an additional backup for cache misses, instead having them
+#   regenerated. The regenration of block structures is a time consuming process. Therefore,
+#   enabling this switch is recommended for Production.
+#   For an example with additional context, see the following forum post:
+#   https://discuss.openedx.org/t/help-please-very-slow-load-time-10-seconds-for-courseware-on-sections-with-several-subsections-and-xblocks/2998/16  pylint: disable=line-too-long,useless-suppression
+# .. toggle_warnings: Depends on `BLOCK_STRUCTURES_SETTINGS['STORAGE_CLASS']` and
+#   `BLOCK_STRUCTURES_SETTINGS['STORAGE_KWARGS']`.
+# .. toggle_use_cases: opt_in
+# .. toggle_creation_date: 2017-02-23
+# .. toggle_tickets: https://github.com/edx/edx-platform/pull/14512
 STORAGE_BACKING_FOR_CACHE = WaffleSwitch(
```

Comment:
@nizarmah Not a blocker for this PR, but:

Similarly to the other toggle above, in order to further reduce the complexity, what do you think about DEPRing this `storage_backing_for_cache` toggle as well?  Is there any reason one would NOT want to have a storage backing? This toggle works on local environment as well as prod, right? Once again, we can announce the removal of this toggle through the DEPR process to double-check.  But there's great value in simply removing these temporary toggles in order to simplify the code.

Reply:
Yeah, I think that it would be a wonderful idea to enable this in the future, by default.

But it would also be helpful to enable `BLOCK_STRUCTURES_SETTINGS['PRUNING_ACTIVE']` as well, just to guarantee that the storage isn't swarmed by block structures 😂 🤷🏼